api/v1alpha1/types.go

```diff
@@ -75,6 +75,10 @@ type CommonThanosFields struct {
 	// +kubebuilder:default:=logfmt
 	// +kubebuilder:validation:Optional
 	LogFormat *string `json:"logFormat,omitempty"`
+	// Enable self monitoring for the Thanos component.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=true
+	EnableSelfMonitor *bool `json:"enableSelfMonitor,omitempty"`
```

Comment:
interested to hear with @saswatamcode thinks, but I think this should be its own struct because there are certain requirements for how service monitors can be created and where they should live to make this useful (thinking rhobs for example).

It could look something like this

```go
type ServiceMonitorConfig struct {
// default true 
Enabled
Labels map[string]string
// Can leave this out for now but it might be required or we can just hard default to current for now
Namespace
// 
}

Reply:
Let's have it in separate struct and maybe as part of commonthanosfields?